coding_style/README.adoc

```diff
@@ -161,6 +165,9 @@ Rationale:: https://yaml.org/type/bool.html[YAML 1.1] allows all variants wherea
   your task will likely need options such as `changed_when:` and maybe `check_mode:`).
 * Anytime `command` or `shell` modules are used, add a comment in the code with justification to help with future maintenance.
 * Use the `| bool` filter when using bare variables (expressions consisting of just one variable reference without any operator) in `when`.
+* No single task file should contain more than 30 tasks.  Complex task files should be broken down into discrete parts.
+* The maximum block depth within a single task file should be 2
+* Bracket notation is preferred over dot notation for value retrieval (e.g. `item[‘key’]` vs. `item.key`)
```

Comment:
I am curious about what benefit the former has over the later. Is it somehow faster to evaluate? Or it's just easier to read?

Reply:
the bracket vs. dot?  There are cases where dot notation can't be used, this rule request links to a good doc about it from NTC: https://github.com/ansible/ansible-lint/issues/2174

It's about consistency more than anything (for lab/home/non-published playbooks I still use dot)

https://github.com/ansible/ansible-lint/issues/2174